src/inventory.cpp

```diff
@@ -1034,6 +1034,20 @@ enchantment inventory::get_active_enchantment_cache( const Character &owner ) co
     return temp_cache;
 }
 
+int inventory::count_item( const itype_id& item_type ) const
```

Comment:
Surely we already have a function that can do this?  Can't you use the code already used to look for crafting ingredients, asking for 9000 of whatever item.  Since there's feedback in the UI about how many items you have when you're short, it must be returning that partial count.

Reply:
I spent some time looking around and while there were a couple I could theoretically use, they all seemed to be either threshold driven (has_components) or, charges vs amounts specific (visitable amount_of, charges_of).  Since I would have to write a wrapper to use those anyways, I figured it would be better to just write a function that did what I wanted.

Reply:
I think a threshold-driven one might be a good option.  As you already pointed out, you don't need to count all when there are many.